### ISSUE TITLE
GLSL ShaderGen: `time_float` and `frame_float` do not have any effect

### ISSUE THREAD

**Issue by hybridherbst** [AcademySoftwareFoundation/MaterialX#1952]
Related to #1951 

I believe the attached file should have some animation based on application time, however it just stays still in MaterialXWebViewer.
[marbleTime.mtlx.zip](https://github.com/user-attachments/files/16389605/marbleTime.mtlx.zip)

<img width="973" alt="image" src="https://github.com/user-attachments/assets/271e3298-b3f4-4eaa-b327-e3f0c1dc078b">

**Comment by hybridherbst** [comment]
Edit: in some applications the time nodes show up as not connected, this one here should work:
[myFile2.mtlx.zip](https://github.com/user-attachments/files/16393512/myFile2.mtlx.zip)

**Comment by kwokcb** [comment]
There is no "play" functionality here nor in the desktop viewer. For the hardware case the `u_frame` shader uniform needs to be updated and bound as mentioned [here](https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/documents/DeveloperGuide/ShaderGeneration.md)

**Comment by hybridherbst** [comment]
Please consider it a feature request :)

**Comment by jstone-lucasfilm** [comment]
Thanks for this feature request, @hybridherbst, and I've just updated the title to clarify that this feature is missing in our GLSL shader generation, which both the Viewer and Graph Editor leverage for rendering

**Comment by jstone-lucasfilm** [comment]
Good point, @hybridherbst, and indeed those updates should be made to the Viewer and Graph Editor when this feature is added.

**Comment by fnRaihanKibria** [comment]
Hello, I'd like to work on this for ASWF Dev Days 2025.

**Comment by jstone-lucasfilm** [comment]
Excellent, thanks @fnRaihanKibria!

**Comment by jstone-lucasfilm** [comment]
Thanks for posting this original issue, @hybridherbst, and thanks to @fnRaihanKibria for the fix in #2389!

If there's still an outstanding compatibility issue with this feature on MacOS, feel free to write up a new GitHub Issue for this, so that we don't lose track.

**Comment by fnRaihanKibria** [comment]
> Thanks for all the work here! However, the file I originally attached to the issue still is not animated – I think this issue here should be reopened.

Are you on a Mac? There is an issue on Macs unfortunately: https://github.com/AcademySoftwareFoundation/MaterialX/issues/2391

**Comment by hybridherbst** [comment]
Oh – yes, I am. The referenced issue seems to have a proposed solution though – are you able to make a PR with that? Thanks!

**Comment by fnRaihanKibria** [comment]
> The PR got merged – I see time working in MaterialXView and MaterialXGraphEditor, but not in the web viewer. Does that need another update?

I believe so. I'm not familiar with it at all and haven't touched it. Might be good to make a new bug ticket?

**Comment by hybridherbst** [comment]
Seems I'm able to fix it myself, I'll make a PR :)
